src/plugins/graph/adornments/movable-line/movable-line.tsx

```diff
@@ -86,198 +96,246 @@ export const MovableLine = observer(function MovableLine(props: IProps) {
+        {domain: xDomain} = xAxis,
+        {domain: yDomain} = yAxis;
+      pointsOnAxes.current[index] = lineToAxisIntercepts(slope, intercept, xDomain, yDomain);
+      const
+        // The coordinates at which the line intersects the axes
+        pixelPtsOnAxes = {
+          pt1: {
+            x: xScale(pointsOnAxes.current[index].pt1.x) / xSubAxesCount,
+            y: yScale(pointsOnAxes.current[index].pt1.y) / ySubAxesCount
+          },
+          pt2: {
+            x: xScale(pointsOnAxes.current[index].pt2.x) / xSubAxesCount,
+            y: yScale(pointsOnAxes.current[index].pt2.y) / ySubAxesCount
           }
-        }
-
-        const
-          // The coordinates at which the line intersects the axes
-          pixelPtsOnAxes = {
-            pt1: {
-              x: xScale(pointsOnAxes.current.pt1.x) / xSubAxesCount,
-              y: yScale(pointsOnAxes.current.pt1.y) / ySubAxesCount
-            },
-            pt2: {
-              x: xScale(pointsOnAxes.current.pt2.x) / xSubAxesCount,
-              y: yScale(pointsOnAxes.current.pt2.y) / ySubAxesCount
-            }
-          };
-        fixEndPoints(lineObject.line);
-        fixEndPoints(lineObject.cover);
-        fixArrow(lineObject.arrowLower, 1);
-        fixArrow(lineObject.arrowUpper, 2);
-        fixHandles(lineObject.handleLower, 1);
-        fixHandles(lineObject.handleUpper, 2);
-        refreshEquation();
-      });
-      return () => disposer();
-    }, [instanceId, layout, pointsOnAxes, lineObject, plotHeight, plotWidth, xScale, yScale, model, model.lines,
-        xAttrName, xSubAxesCount, xAxis, yAttrName, ySubAxesCount, yAxis, xRange, yRange, positionEquation,
-        equationContainerSelector, subPlotKey, instanceKey, calculateHandlePosition, annotationLocationSetter]
-  );
+        };
+      lineObject.line && fixEndPoints(lineObject.line, pixelPtsOnAxes);
+      lineObject.cover && fixEndPoints(lineObject.cover, pixelPtsOnAxes);
+      lineObject.arrowLower && fixArrow(lineObject.arrowLower, 1, pixelPtsOnAxes);
+      lineObject.arrowUpper && fixArrow(lineObject.arrowUpper, 2, pixelPtsOnAxes);
+      lineObject.handleLower && fixHandles(lineObject.handleLower, 1, pixelPtsOnAxes, lineModel, lineObject.key);
+      lineObject.handleUpper && fixHandles(lineObject.handleUpper, 2, pixelPtsOnAxes, lineModel, lineObject.key);
+      updateClasses(lineObject.line, lineObject.key);
+      refreshEquation(slope, intercept, lineModel, index, lineObject.key);
+    });
+  }, [annotationLocationSetter, calculateHandlePosition, instanceId, layout, model.isVisible, model.lines, plotHeight,
+      plotWidth, positionEquation, updateClasses, xAttrName, xAxis, xScale, xSubAxesCount, yAttrName, yAxis, yScale,
+      ySubAxesCount]);
+
+  // Refresh the scale copies
+  useEffect(() => {
+    const disposer = autorun(() => {
+      xScaleCopy.current = xScale.copy();
+      yScaleCopy.current = yScale.copy();
+    });
+    return () => disposer();
```

Comment:
There's no reason for the `autorun` here because there's nothing observable:
```suggestion
    xScaleCopy.current = xScale.copy();
    yScaleCopy.current = yScale.copy();
```
The effect will run when either of its dependencies changes. It's still a bit of a mystery why this is necessary, but apparently the scales are updated elsewhere after the initial render but before other effects are run.